### PR TITLE
twister: check if qemu is executable

### PR DESCRIPTION
**Pull request by wangnuannuan**
When the qemu binary file doesn't exist or it is unexecutable, cmake will not throw error message, also the twister can't
detect this error, then the twister will hang forever

https://github.com/zephyrproject-rtos/zephyr/issues/35386

Signed-off-by: Jingru Wang <jingru@synopsys.com>

### CODE REVIEW

**Review by galak — COMMENTED** [review]
Feels like this should be an error if the qemu binary isn't executable.

File: scripts/pylib/twister/twisterlib.py [hunk 1]
```diff
@@ -1075,6 +1075,12 @@ def _thread(handler, timeout, outdir, logfile, fifo_fn, pid_fn, results, harness
 
     def handle(self):
         self.results = {}
+        qemu_executable = self.instance.platform.filter_data.get("QEMU")
```
Comment (hakehuang) [inline scripts/pylib/twister/twisterlib.py:1078]: in what situation quma binary is not exist or not executable?

**Review by hakehuang — APPROVED** [review]
just wonder why the qemu binary is missing or not executable

File: scripts/pylib/twister/twisterlib.py (hunk 1 above)
Comment (wangnuannuan) [reply scripts/pylib/twister/twisterlib.py:1078]: Previously, to get the SDK, I copied it directly from another machine. In which case  twister would hang forever without any error message, I spent a lot of time to find out the root cause ( I didn't try to run case with west), finanlly I found  qemu binary didn't work. So I think it's necessay to verify the qemu binary in twister

Comment (hakehuang) [reply scripts/pylib/twister/twisterlib.py:1078]: in this case, it is a false operation, it is better to return error, not skip

**Review by tejlmand — APPROVED** [review]

**Review by hakehuang — APPROVED** [review]
my concerns are fixed, looks good to me.

**Comment by wangnuannuan** [comment]
https://github.com/zephyrproject-rtos/zephyr/pull/36257 also can handle this problem

**Pull request closed without merging**